### PR TITLE
Multi/rework v1.0.9

### PR DESCRIPTION
**Pull request by alinmigea**
@plentymarkets/team-multichannel-koboldmaki

### CODE REVIEW

File: src/Generator/IdealoDE.php [hunk 1]
```diff
@@ -190,17 +200,11 @@ protected function generatePluginContent($elasticSearch, array $formatSettings =
...
-                                ]);
-
-                                continue;
-                            }
-
+                            // Add the new variation to the grouper
                             $variations[] = $variation;
+
+                            // New line will be printed in the CSV
+                            $limit += 1;
```
Comment (Desentyer) [inline src/Generator/IdealoDE.php]: In the next format please use `$limit++` =D

File: src/Helper/PriceHelper.php [hunk 2]
```diff
@@ -41,18 +50,15 @@ public function getPriceList($variation, KeyValue $settings):array
...
-         * SalesPriceSearchRequest $salesPriceSearchRequest
-         */
-        $salesPriceSearchRequest = pluginApp(SalesPriceSearchRequest::class);
-        if($salesPriceSearchRequest instanceof SalesPriceSearchRequest)
+        if($this->salesPriceSearchRequest instanceof SalesPriceSearchRequest)
         {
-            $salesPriceSearchRequest->variationId = $variation['id'];
-            $salesPriceSearchRequest->referrerId = $settings->get('referrerId');
+            $this->salesPriceSearchRequest->variationId = $variation['id'];
+            $this->salesPriceSearchRequest->referrerId = $settings->get('referrerId');
```
Comment (Desentyer) [inline src/Helper/PriceHelper.php]: Add a fallback value pls.

File: src/Helper/PriceHelper.php [hunk 3]
```diff
@@ -21,13 +21,22 @@ class PriceHelper
...
+     * @param SalesPriceSearchRequest $salesPriceSearchRequest
      */
-    public function __construct(SalesPriceSearchRepositoryContract $salesPriceSearchRepository)
+    public function __construct(
+        SalesPriceSearchRepositoryContract $salesPriceSearchRepositoryContract,
+        SalesPriceSearchRequest $salesPriceSearchRequest)
     {
-        $this->salesPriceSearchRepository = $salesPriceSearchRepository;
+        $this->salesPriceSearchRepository = $salesPriceSearchRepositoryContract;
+        $this->salesPriceSearchRequest = $salesPriceSearchRequest;
```
Comment (Desentyer) [inline src/Helper/PriceHelper.php]: Has team Item extended the repositories with clearCriteria functions? I have only extended the stock repository

File: src/Generator/IdealoDE.php (hunk 1 above)
Comment (alinmigea) [inline src/Generator/IdealoDE.php]: yep :)

**Pull request merged**